Add Redeem Script page (#1089)
decred/dcrdocs: docs/proof-of-stake/redeem-script.md

# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Redeem Script

---

> When users sign up with a VSP, both the user and the VSP jointly create a 1-of-2 multisignature script. This script is known as the redeem script.

Author: Agreed, added the link and spelled out the acronym

The redeem script involves both a public key from the user and a public key from the VSP so that both can vote and revoke the ticket. When the user buys a new ticket, the voting rights are given to a hash of that redeem script.

---

## Why users must backup their scripts

At first, when the user just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.

> The VSP handles voting and revoking for the user, but if it goes offline is up to the user to vote or revoke, which means the wallet needs to know the redeem script.

Author: Maybe "...if it goes offline, it is up to the user to vote using the remaining active tickets or revoke those that have expired. In order to do this, the user's wallet needs to know the redeem script."

> The VSP handles voting and revoking for the user, but if it goes offline is up to the user to vote or revoke, which means the wallet needs to know the redeem script.

Reviewer: Your suggested wording is clearer. Let's use it.

Currently, users can't reconstruct these scripts from their own seed, this is why they must [backup their redeem scripts](/wallets/decrediton/using-decrediton/#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.

---

## What happens if the script is lost?

User who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets. This means their funds will remain locked.

---

## Recovery methods

> Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred’s block explorer, [dcrdata](https://explorer.dcrdata.org).

Author: I've left the instructions on the Redeem Script page and added a brief explanation:

> When a ticket votes or is revoked, that redeem script is revealed as a part of the process. Redeem scripts that have been used at least once to vote or revoke a ticket can be [recovered using Decred's block explorer](/proof-of-stake/redeem-script/#recovery-methods).

Almost forgot about the boxes

> Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred’s block explorer, [dcrdata](https://explorer.dcrdata.org).

Reviewer: Lgtm.

# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Redeem Script

---

> When users sign up with a [Voting Service Provider](/glossary/#voting-service-provider) (VSP), both the user and the VSP jointly create a 1-of-2 multisignature script. This script is known as the redeem script.

Reviewer 2: Link to `.md` file

> The redeem script involves both a public key from the user and a public key from the VSP so that both can vote and revoke the ticket. When the user buys a new ticket, the voting rights are given to a hash of that redeem script.

Reviewer 2: Two things here:

- A ticket can only be voted **or** revoked, not both.
- I think these two sentences actually make more sense if the order is reversed.

Suggest changing to

> When a VSP user buys a new ticket, the voting rights are given to a hash of the redeem script. The redeem script involves both a public key from the user and a public key from the VSP, so either are able to vote or revoke the purchased tickets.

---

## Why users must backup their scripts

> At first, when the user just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.

Reviewer 2: At first, when the user has just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.
```suggestion
At first, when a user has just signed up to a VSP, the redeem script is not known by anyone except the user and the VSP. When a ticket votes or is revoked, that redeem script is revealed on-chain as a part of the process.
```

> The VSP handles voting and revoking for the user, but if it goes offline it is up to the user to vote or revoke the tickets, which means the wallet needs to know the redeem script.

Reviewer 2: ```suggestion
The VSP handles voting and revoking for the user, but if it goes offline it is up to the user to vote or revoke the tickets, which means the users wallet also needs to know the redeem script.
```

Currently, users can't reconstruct these scripts from their own seed. This is why they must [backup their redeem scripts](/wallets/decrediton/using-decrediton/#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.

---

## What happens if the script is lost?

A user who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets. This means their funds will remain locked.

---

## Recovery methods

Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred's block explorer, [dcrdata](https://dcrdata.org).

> 1. Look for one of your vote or revoke transactions. This can be done creating a new wallet using your seed phrase, and checking your past tickets.

Reviewer 2: Numbered bullets should all use `1.` so they are automatically incremented. This removes the need to update all lines if bullets are added/removes in future.

# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Redeem Script

---

When users sign up with a [Voting Service Provider](../glossary.md#voting-service-provider) (VSP), both the user and the VSP jointly create a 1-of-2 multisignature script. This script is known as the redeem script.

When a VSP user buys a new ticket, the voting rights are given to a hash of the redeem script. The redeem script involves both a public key from the user and a public key from the VSP, so either are able to vote or revoke the purchased tickets.

---

## Why users must backup their scripts

At first, when a user has just signed up to a VSP, the redeem script is not known by anyone except the user and the VSP. When a ticket votes or is revoked, that redeem script is revealed on-chain as a part of the process.

> The VSP handles voting and revoking for the user, but if it goes offline, it is up to the user to vote using the remaining active tickets or revoke those that have expired. In order to do this, the users wallet also needs to know the redeem script.

Reviewer 2: ```suggestion
The VSP handles voting and revoking for the user, but if it goes offline, it is up to the user to vote the remaining active tickets or revoke those that have expired. In order to do this, the users wallet also needs to know the redeem script.
```

> When registering with a VSP, the user wallet will automatically store a copy of the redeem script, but currently, users can't reconstruct these scripts from their own seed. This is why they must [backup their redeem scripts](../wallets/decrediton/using-decrediton.md#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.

Reviewer 2: ```suggestion
When registering with a VSP, the users wallet will automatically store a copy of the redeem script, but currently, users can't reconstruct these scripts from their own seed. This is why they must [backup their redeem scripts](../wallets/decrediton/using-decrediton.md#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.
```

---

## What happens if the script is lost?

A user who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets.

Users risk having their funds locked if the following conditions are met:

> - The user has not backed up their redeem script as the documentation and software instructs

Reviewer 2: ```suggestion
- The user has not backed up their redeem script
```
